kairon/importer/validator/file_validator.py

```diff
@@ -194,12 +206,27 @@ def verify_intents_in_stories(self, raise_exception: bool = True):
                     raise AppException(msg)
                 intents_mismatched.append(msg)
 
+        if multiflow_stories_intent:
+            for intent in multiflow_stories_intent:
+                if intent not in self.domain.intents and intent not in DEFAULT_INTENTS:
+                    msg = f"The intent '{intent}' is used in your multiflow_stories, but it is not listed in " \
+                          f"the domain file. You should add it to your domain file!"
+                    if raise_exception:
+                        raise AppException(msg)
+                    intents_mismatched.append(msg)
+
         for intent in self.domain.intents:
-            if intent not in stories_intents and intent not in DEFAULT_INTENTS:
-                msg = f"The intent '{intent}' is not used in any story."
-                if raise_exception:
-                    raise AppException(msg)
-                intents_mismatched.append(msg)
+            msg = f"The intent '{intent}' is not used in any story."
```

Comment:
simplify this
use sets and join intents from multiflow as well as normal stories